lib/govuk_publishing_components/presenters/page_with_step_by_step_navigation.rb

```diff
@@ -24,7 +30,8 @@ def show_header?
       end
 
       def show_related_links?
-        step_navs.any? && (step_navs.count < 5 || active_step_by_step?)
+        return true if active_step_by_step? && (step_navs.any? || related_to_step_navs.any?)
```

Comment:
You should only need to check for `active_step_by_step?` here. `active_step_by_step?` is already implicitly doing the `step_navs.any? || related_to_step_navs.any?` check